src/components/HdResponsive.vue

```diff
@@ -18,42 +16,46 @@ export default {
     return {
       matches: {},
       indeterminate: !matchMediaAvailable,
+      listeners: {},
     };
   },
   watch: {
     breakpoints() {
-      this.setResponsiveFlags();
+      this.setListeners();
     },
   },
-  created() {
-    this.setResponsiveFlags();
-  },
-  mounted() {
-    this.addResizeEvents();
-  },
-  beforeDestroy() {
-    this.removeResizeEvents();
+  beforeMount() {
+    this.setListeners();
   },
   methods: {
-    setResponsiveFlags() {
+    /* istanbul ignore next */
+    updateMatch(key, value) {
+      this.matches[key] = value;
+      this.$forceUpdate();
```

Comment:
Could you briefly describe the issue and why you went for `$forceUpdate`?

Reply:
Because it's not re-rendering at the matches change.